examples/SampleApp/src/hooks/usePaginatedPinnedMessages.ts

```diff
@@ -0,0 +1,105 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+
+import { AppContext } from '../context/AppContext';
+
+import type { Channel, MessageResponse } from 'stream-chat';
+
+import type {
+  LocalAttachmentType,
+  LocalChannelType,
+  LocalCommandType,
+  LocalEventType,
+  LocalMessageType,
+  LocalReactionType,
+  LocalUserType,
+} from '../types';
+
+export const usePaginatedPinnedMessages = (
+  channel: Channel<
+    LocalAttachmentType,
+    LocalChannelType,
+    LocalCommandType,
+    LocalEventType,
+    LocalMessageType,
+    LocalReactionType,
+    LocalUserType
+  >,
+) => {
+  const { chatClient } = useContext(AppContext);
+  const offset = useRef(0);
+  const hasMoreResults = useRef(true);
+  const queryInProgress = useRef(false);
+  const [loading, setLoading] = useState(true);
+  const [messages, setMessages] = useState<
+    MessageResponse<
+      LocalAttachmentType,
+      LocalChannelType,
+      LocalCommandType,
+      LocalMessageType,
+      LocalReactionType,
+      LocalUserType
+    >[]
+  >([]);
+
+  const fetchPinnedMessages = async () => {
+    if (queryInProgress.current) {
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      queryInProgress.current = true;
+
+      offset.current = offset.current + messages.length;
+
+      if (!hasMoreResults.current) {
+        queryInProgress.current = false;
+        return;
+      }
+
+      // TODO: Use this when support for attachment_type is ready.
+      const res = await chatClient?.search(
+        {
+          cid: { $in: [channel.cid] },
+        },
+        { pinned: true },
+        {
+          limit: 10,
+          offset: offset.current,
+        },
+      );
+
+      const newMessages = res?.results.map((r) => r.message);
+
+      if (!newMessages) {
+        queryInProgress.current = false;
+        return;
+      }
+
+      setMessages((existingMessages) => existingMessages.concat(newMessages));
+
+      if (newMessages.length < 10) {
+        hasMoreResults.current = false;
+      }
+    } catch (e) {
+      // do nothing;
```

Comment:
Might it be worth throwing a warning to the console if something goes wrong, maybe? Not important in an example like this though, I suppose, but doing nothing if it fails seems like it could be a bit confusing to the developer if something goes wrong and they have to track it down themselves

Reply:
We can do it, but currently, it's not handled anywhere. I will have a look into this with a different issue. What say @vishalnarkhede, is it worth adding this in this PR?

Reply:
Yeah let's do it. So although its a sample app and not SDK, customers still copy-paste these hooks into their codebase plenty of times. So worth adding a proper warning here.